Library/Homebrew/cli/named_args.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+require "delegate"
+
+module Homebrew
+  module CLI
+    class NamedArgs < SimpleDelegator
+      def initialize(*args, override_spec: nil, force_bottle: false, flags: [])
+        @args = args
+        @override_spec = override_spec
+        @force_bottle = force_bottle
+        @flags = flags
+
+        __setobj__(@args)
+      end
+
+      def to_formulae
+        @to_formulae ||= downcased_unique_named.map do |name|
+          Formulary.factory(name, spec, force_bottle: @force_bottle, flags: @flags)
+        end.uniq(&:name).freeze
+      end
+
+      def to_formulae_and_casks
+        @to_formulae_and_casks ||= begin
+          formulae_and_casks = []
+
+          downcased_unique_named.each do |name|
+            formulae_and_casks << Formulary.factory(name, spec)
+          rescue FormulaUnavailableError
+            begin
+              formulae_and_casks << Cask::CaskLoader.load(name)
+            rescue Cask::CaskUnavailableError
+              raise "No available formula or cask with the name \"#{name}\""
+            end
+          end
+
+          formulae_and_casks.freeze
+        end
+      end
+
+      def to_resolved_formulae
+        @to_resolved_formulae ||= downcased_unique_named.map do |name|
+          Formulary.resolve(name, spec: spec(nil), force_bottle: @force_bottle, flags: @flags)
+        end.uniq(&:name).freeze
+      end
+
+      def to_resolved_formulae_to_casks
+        @to_resolved_formulae_to_casks ||= begin
+          resolved_formulae = []
+          casks = []
+
+          downcased_unique_named.each do |name|
+            resolved_formulae << Formulary.resolve(name, spec: spec(nil), force_bottle: @force_bottle, flags: @flags)
+          rescue FormulaUnavailableError
+            begin
+              casks << Cask::CaskLoader.load(name)
+            rescue Cask::CaskUnavailableError
+              raise "No available formula or cask with the name \"#{name}\""
+            end
+          end
+
+          [resolved_formulae.freeze, casks.freeze].freeze
+        end
+      end
+
+      def to_formulae_paths
+        @to_formulae_paths ||= downcased_unique_named.map do |name|
+          Formulary.path(name)
+        end.uniq.freeze
+      end
+
+      def to_casks
```

Comment:
Based on the naming, I would expect this to return what is currently returned by `to_loaded_casks`.